workspaces/liferay-partner-workspace/client-extensions/liferay-partner-etc-spring-boot/src/main/java/com/liferay/partner/service/CloudFunctionsWebService.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.partner.service;
+
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.IdTokenCredentials;
+import com.google.auth.oauth2.IdTokenProvider;
+import com.google.common.io.CharStreams;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import java.nio.charset.StandardCharsets;
+
+import org.json.JSONObject;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Felipe Franca
+ */
+@Component
+public class CloudFunctionsWebService {
```

Comment:
@abfranca I think this will be simplier and easier to maintain if we import and use the google supported client library for cloud functions (https://cloud.google.com/java/docs/reference/google-cloud-functions/2.33.0/overview)

Reply:
Also, to keep things consistent it would be good to name it GoogleCloudFnctionsWebService. Both of these are similar to what Amos is doing here (https://github.com/liferay/liferay-portal/blob/master/workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/service/GoogleCloudStorageWebService.java)